### PR TITLE
APR-1656 AssessorPageReviewOutcome Table

### PR DESCRIPTION
**Pull request by MakCom**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	[SequenceNumber] [int] NOT NULL,
+	[SectionNumber] [int] NOT NULL,
+	[PageId] [nvarchar](50) NOT NULL,
+	[Assessor1UserId] NVARCHAR(256) NULL,     
+	[Assessor1ReviewStatus] NVARCHAR(20) NULL, 
+    [Assessor1ReviewComment] NVARCHAR(MAX) NULL,
+	[Assessor2UserId] NVARCHAR(256) NULL, 
+	[Assessor2ReviewStatus] NVARCHAR(20) NULL,
+	[Assessor2ReviewComment] NVARCHAR(MAX) NULL,
+	[CreatedAt] DATETIME2 NOT NULL,
```
Comment (martin-hornsby) [inline src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql:13]: Add default for CreatedAt

ALTER TABLE [dbo].AssessorPageReviewOutcome ADD  DEFAULT (getdate()) FOR [CreatedAt]

File: src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿CREATE TABLE [dbo].[AssessorPageReviewOutcome](
+	[Id] [uniqueidentifier] NOT NULL,
+	[ApplicationId] [uniqueidentifier] NOT NULL,
+	[SequenceNumber] [int] NOT NULL,
+	[SectionNumber] [int] NOT NULL,
+	[PageId] [nvarchar](50) NOT NULL,
+	[Assessor1UserId] NVARCHAR(256) NULL,
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql:7]: Do we also need the assessor name?

File: src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    [UpdatedBy] NVARCHAR(256) NULL, 
+ CONSTRAINT [PK_AssessorPageReviewOutcome] PRIMARY KEY CLUSTERED 
+(
+	[Id] ASC
+)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY], 
+) ON [PRIMARY] TEXTIMAGE_ON [PRIMARY]
+GO
+
+ALTER TABLE [dbo].AssessorPageReviewOutcome ADD  DEFAULT (newid()) FOR [Id]
+GO
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql:25]: We'll need some indexes.  Application Id as a minimum.

File: src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql (hunk 1 above)
Comment (MakCom) [reply src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql:13]: For all other tables we are adding CreatedAt (and UpdatedAt) in the ApplyRepository.
Do you still want me to add it the DB?
And for all other cases GETUTCDATE() is used.
Which one I should use GETUTCDATE() or GETDATE() ?

File: src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql (hunk 2 above)
Comment (MakCom) [reply src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql:7]: No. I don’t think that we need assessor name. We don’t need to display it anywhere.
We need IDs only to identify the records and update the correct ones.

File: src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql (hunk 3 above)
Comment (MakCom) [reply src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql:25]: The querying of the data will be by:
GetAllStatuses()
-	ApplicationId, (Assessor1UserId or Assessor2UserId)
GetStatusesPerSection() – to work out SectionStatus (might not be needed if use above and group them) 
-	ApplicationId, SequenceNumber, SectionNumber (Assessor1UserId or Assessor2UserId)
GetPageStatus()
-	ApplicationId, PageId, (Assessor1UserId or Assessor2UserId)

So, which fields should I index?

Comment (MichaelYoung1981) [reply src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql:25]: I think an index on ApplicationId will be sufficient.

File: src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql (hunk 2 above)
Comment (MichaelYoung1981) [reply src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql:7]: 👍

File: src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql (hunk 3 above)
Comment (MakCom) [reply src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql:31]: index IX_AssessorPageReviewOutcome_ApplicationId added

File: src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql (hunk 1 above)
Comment (MakCom) [reply src/SFA.DAS.ApplyService.Database/Tables/AssessorPageReviewOutcome.sql:13]: ALTER TABLE [dbo].AssessorPageReviewOutcome ADD DEFAULT (getutcdate()) FOR [CreatedAt] added

**Pull request merged**